core/core-impl/src/main/java/ai/timefold/solver/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -1026,7 +1030,39 @@ public long getGenuineVariableCount(Solution_ solution) {
         return result.longValue();
     }
 
-    public long getMaximumValueCount(Solution_ solution) {
+    /**
+     * @param solution never null
+     * @return {@code >= 0}
+     */
+    public long getApproximateValueCount(Solution_ solution) {
+        Set<GenuineVariableDescriptor<Solution_>> genuineVariableDescriptorSet =
```

Comment:
IMO this is precisely the kind of code that benefits from `var` the most. We have `Set` in the var name, we have `Set` in the method call on the right, no need to repeat the entire type on the left. (Even if it means putting the generic type directly to the diamond of `IdentityHashMap`.)